core/src/components/header/header.tsx

```diff
@@ -24,6 +36,76 @@ export class Header implements ComponentInterface {
    */
   @Prop() translucent = false;
 
+  async componentDidLoad() {
+    // Determine if the header can collapse
+    const canCollapse = (this.collapse && getIonMode(this) === 'ios') ? this.collapse : false;
+
+    const tabs = this.el.closest('ion-tabs');
+    const page = this.el.closest('ion-app,ion-page,.ion-page,page-inner');
+    const contentEl = tabs ? tabs.querySelector('ion-content') : page!.querySelector('ion-content');
+
+    if (canCollapse) {
+      await this.setupCollapsableHeader(contentEl, (tabs) ? tabs : page!);
+    }
+  }
+
+  componentDidUnload() {
+    if (this.contentScrollCallback) {
+      this.scrollEl.removeEventListener('scroll', this.contentScrollCallback);
```

Comment:
scrollEl might be undefined, once any is removed, ts will complain, wrap around if(this.scrollEl)